native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/NativeExtension.java

```diff
@@ -129,24 +133,19 @@ public void afterProjectsRead(MavenSession session) {
             withPlugin(build, "native-maven-plugin", nativePlugin -> {
                 String target = build.getDirectory();
                 String testIdsDir = testIdsDirectory(target);
-                boolean isAgentEnabled = isAgentEnabled(session, nativePlugin);
-                String selectedOptionsName = getSelectedOptionsName(session);
-                Xpp3Dom x = Xpp3DomParser.getTagByName((Xpp3Dom) nativePlugin.getConfiguration(), "agent");
-                System.out.println("\n\n\n\n\n\n*****************************************************************************\n\n\n\n\n\n");
-                assert x != null;
-                System.out.println(x.getName() + " ------- " + x.getParent().getName());
+                AgentConfiguration agent = AgentUtils.collectAgentProperties(session, (Xpp3Dom) nativePlugin.getConfiguration());
 
                 // Test configuration
                 withPlugin(build, "maven-surefire-plugin", surefirePlugin -> {
                     configureJunitListener(surefirePlugin, testIdsDir);
-                    if (isAgentEnabled) {
-                        List<String> agentOptions = getAgentOptions(nativePlugin, Context.test, selectedOptionsName);
+                    if (agent != null) {
```

Comment:
It would be better if `AgentUtils.collectAgentProperties()` always return a non-null `AgentConfiguration` instance with the enabled flag set so that you can do:

```java
if (agent.isEnabled()) { ... }
```